### PR TITLE
Support external repos for push items

### PR DESCRIPTION
**Pull request by midnightercz**
When push items are populated by ET, they include both repos
and external_repos. In the case external repos are set, they
should be prioritized and used instead of 'repos'

### CODE REVIEW

File: pubtools/_quay/push_docker.py [hunk 1]
```diff
@@ -229,7 +231,12 @@ def check_repos_validity(self, push_items):
         """
         repos = []
         for item in push_items:
-            repos += item.repos.keys()
+            LOG.info("ITEM %s", item)
```
Comment (emilyzheng) [inline pubtools/_quay/push_docker.py:234]: Debugging log?

**Review by emilyzheng — APPROVED** [review]

File: pubtools/_quay/push_docker.py (hunk 1 above)
Comment (midnightercz) [reply pubtools/_quay/push_docker.py:234]: right, I need to remove this

File: pubtools/_quay/push_docker.py [hunk 2]
```diff
@@ -152,6 +144,16 @@ def get_docker_push_items(self):
 
         return docker_push_items
 
+    @log_step("Translate docker push items")
+    def filter_unrelated_repos(self, push_items):
+        """Remove item repos from tag mapping if external_repos is set for push item."""
```
Comment (querti) [inline pubtools/_quay/push_docker.py:149]: Could you write a full docstring including the arguments?

File: pubtools/_quay/push_docker.py [hunk 3]
```diff
@@ -152,6 +144,16 @@ def get_docker_push_items(self):
...
+        for push_item in push_items:
+            if push_item.external_repos:
+                #  if external repos is defined, push items was populated by ET
+                #  item.repos have to be remove from tags mapping
```
Comment (querti) [inline pubtools/_quay/push_docker.py:153]:
```suggestion
                #  item.repos have to be removed from tags mapping
```

File: tests/test_push_docker.py [hunk 4]
```diff
@@ -390,14 +399,18 @@ def test_check_repos_validity_success(
...
+            container_push_item_correct_repos,
+            container_signing_push_item,
+        ]
     )
 
     mock_get_target_info.assert_called_once_with("target_stage_quay")
-    assert mock_get_repo_metadata.call_count == 2
+    assert mock_get_repo_metadata.call_count == 3
     mock_get_repo_metadata.call_args_list[0] == mock.call("namespace/repo1")
     mock_get_repo_metadata.call_args_list[1] == mock.call("namespace/repo2")
```
Comment (querti) [inline tests/test_push_docker.py:412]: Can you add checks for the extra calls?

File: tests/test_push_docker.py [hunk 5]
```diff
@@ -830,21 +850,28 @@ def test_push_docker_full_success(
 
     mock_get_docker_push_items.assert_called_once_with()
     mock_get_docker_push_items.assert_called_once_with()
-    mock_check_repos_validity.assert_called_once_with([container_multiarch_push_item])
-    mock_generate_backup_mapping.assert_called_once_with([container_multiarch_push_item])
+    mock_check_repos_validity.assert_called_once_with(
```
Comment (querti) [inline tests/test_push_docker.py:853]: Could you add a check if filter_unrealted_repos was called?

File: pubtools/_quay/push_docker.py [hunk 6]
```diff
@@ -152,6 +144,16 @@ def get_docker_push_items(self):
 
         return docker_push_items
 
+    @log_step("Translate docker push items")
+    def filter_unrelated_repos(self, push_items):
```
Comment (querti) [inline pubtools/_quay/push_docker.py:148]: Could you add a separate test for this new method?

**Review by querti — CHANGES_REQUESTED** [review]

File: tests/test_push_docker.py (hunk 5 above)
Comment (midnightercz) [reply tests/test_push_docker.py:853]: it's not mocked so it has to be called

**Review by querti — APPROVED** [review]

**Pull request merged**